src/cli/lint/external.ts

```diff
@@ -32,24 +50,32 @@ const lintSerially = async (input: Input) => {
 export const externalLint = async (input: Input) => {
   log.newline();
 
-  const lint =
-    // `--debug` implies `--serial`.
-    input.debug || input.serial ? lintSerially : lintConcurrently;
+  // `--debug` implies `--serial`.
+  const isSerial = input.debug || input.serial;
 
-  const { eslint, prettier, tscOk } = await lint(input);
+  const lint = isSerial ? lintSerially : lintConcurrently;
 
-  log.newline();
+  const tscOutputStream = new StreamInterceptor();
+  tscOutputStream.pipe(input.tscOutputStream ?? process.stdout);
+
+  const { eslint, prettier, tscOk } = await lint({ ...input, tscOutputStream });
 
   if (eslint.ok && prettier.ok && tscOk) {
     return;
   }
 
+  if (!isSerial) {
+    // stdio from worker threads can lag a bit, so we wait for a little while.
```

Comment:
This is annoying me but I can't think of anywhere we're missing a flush or an `await`. Is this just data bouncing around streams and `postMessage` that haven't completely settled yet?

Reply:
Thanks for calling me on this, I think I've fixed it proper in de883b05935975a73f45bdb92b01815433469d45.